dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/BeamSqlPrimitive.java

```diff
@@ -67,34 +67,34 @@ public T getValue() {
   }
 
   public long getLong() {
-    return (Long) getValue();
+    return Long.valueOf(getString());
```

Comment:
if I have short/long type and have to convert it to double then in these kind of cases the casting would not work.